var/spack/repos/builtin/packages/astyle/package.py

```diff
@@ -11,8 +11,13 @@ class Astyle(Package):
     def install(self, spec, prefix):
 
         with working_dir('src'):
+            # unfortunately we need to edit the makefile in place to set compiler:
+            make_file = join_path(self.stage.source_path,'build','gcc','Makefile')
+            filter_file(r'^CXX\s*=.*', 'CXX=%s' % spack_cxx, make_file)
```

Comment:
Maybe you're right, I have too many languages jammed into my brain right now. Maybe I'm thinking of Perl?